### PR TITLE
Automation extrinisic pallet ordering checking

### LINKED ISSUE #577: Automation of entrinsic and pallet order check
### Context
We'd like to perform automated checks similar to [polkadot](https://github.com/paritytech/polkadot/blob/master/.github/workflows/extrinsic-ordering-check-from-bin.yml).

It should summarize the changes and remind us to bump the `transaction_version` if required, like [here](https://github.com/paritytech/polkadot/issues/5040#issuecomment-1063292396).

### Task
- [x] write/integrate the auto-checking script into GA
- [x] trigger the check upon release, ideally automatically

### Acceptance criteria
the script is integrated and the checks can be successfully carried out upon release.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by chenzongxiong**
Currently, we can compare metadata of parachain on the production with the one to be released and automatically comment the corresponding release checklist. 

**NOTE:**
_release checklist_ issue **MUST** be created before running `create release draft` action, otherwise, metadata comparison cannot perform correctly 

Check the following results, we can confirm that this action can perform correctly
https://github.com/litentry/litentry-parachain/actions/runs/2743075428

[Litentry-parachain v0.9.7 Release checklist](https://github.com/litentry/litentry-parachain/issues/578)

### CODE REVIEW

File: .github/workflows/extrinsic-ordering-check-from-bin.yml [hunk 1]
```diff
@@ -1,29 +1,18 @@
 name: Release - Extrinsic Ordering Check
 
 on:
+  workflow_dispatch:
   workflow_run:
-    workflows: ["Create release draft", "Issue stale bot"]
+    workflows: ["Create release draft"]
     branches:
```
Comment (Kailai-Wang) [inline .github/workflows/extrinsic-ordering-check-from-bin.yml]: This won't work. The workflow "Create release draft" will (almost) always run on release branch, but probably never on `dev` branch.

I think we can simply delete the branches directive.

File: .github/workflows/extrinsic-ordering-check-from-bin.yml [hunk 2]
```diff
@@ -33,35 +22,33 @@ jobs:
 
       - name: Prepare output and compare the metadata
         run: |
-          CHAIN=${CHAIN:-litmus}
           BASE_URL=wss://parachain-eu-0.staging.litentry.io
```
Comment (Kailai-Wang) [inline .github/workflows/extrinsic-ordering-check-from-bin.yml]: This is incorrect if I understand it correctly, we'd need to compare the live chain and the image that is going to be released. Therefore we need to start the parachain network locally (or only the parachain image if that works) beforehand using the to-be-released docker image

You could refer to [polkadot](https://github.com/paritytech/polkadot/blob/9804117d3e676aace31ed39569a4e6aec75956de/.github/workflows/release-20_extrinsic-ordering-check-from-bin.yml#L60) or [acala](https://github.com/AcalaNetwork/Acala/blob/6334e0f5547bfc5aeb30a19a94730ad582cca593/.github/workflows/extrinsic-ordering-check-from-bin.yml#L50)

File: .github/workflows/extrinsic-ordering-check-from-bin.yml [hunk 3]
```diff
@@ -33,35 +22,33 @@ jobs:
...
-          CMD="docker run --pull always --network host jacogr/polkadot-js-tools metadata $REF_URL $BASE_URL"
-          echo -e "Running:\n$CMD"
-          $CMD >> output.txt
-          sed -z -i 's/\n\n/\n/g' output.txt
-          cat output.txt | egrep -n -i ''
-          SUMMARY=$(./scripts/extrinsic-ordering-filter.sh output.txt)
-          echo -e $SUMMARY
-          echo -e $SUMMARY >> output.txt
-
+          chains=( "litmus" "litentry" )
```
Comment (Kailai-Wang) [inline .github/workflows/extrinsic-ordering-check-from-bin.yml]: Can we add rococo too?

File: .github/workflows/extrinsic-ordering-check-from-bin.yml [hunk 4]
```diff
@@ -33,35 +22,33 @@ jobs:
...
 
       - name: Save output as artifact
         uses: actions/upload-artifact@v3
         with:
-          name: output-${{ env.CHAIN }}
+          name: output
           path: |
-            output.txt
+            output-litmus.txt
+            output-litentry.txt
```
Comment (Kailai-Wang) [inline .github/workflows/extrinsic-ordering-check-from-bin.yml]: Ideally this output summary can be posted as comments in the release checklist ticket (like polkadot), but it's not a must and we can do it in a separate PR later.

File: .github/workflows/extrinsic-ordering-check-from-bin.yml [hunk 5]
```diff
@@ -1,29 +1,18 @@
 name: Release - Extrinsic Ordering Check
```
Comment (Kailai-Wang) [inline .github/workflows/extrinsic-ordering-check-from-bin.yml]: We can simply call it "Extrinsic Ordering Check"

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]
The workflow is fine, but the main problem is the comparing URL

File: .github/workflows/extrinsic-ordering-check-from-bin.yml (hunk 4 above)
Comment (chenzongxiong) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: Any special requirements about created Issue. I found some reference from `release draeft action` you wrote.

Comment (Kailai-Wang) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: The release issue (checklist) should have been created earlier already, when we push tags on the release branch. So we might need to search for that specific ticket and attach comment to it.

We can do this task later, it's not urgent

File: .github/workflows/extrinsic-ordering-check-from-bin.yml (hunk 1 above)
Comment (chenzongxiong) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: Compare `rococo`, `litmus` and  `litentry` on production endpoint with the parachain to be released endpoint.

File: .github/workflows/extrinsic-ordering-check-from-bin.yml (hunk 3 above)
Comment (chenzongxiong) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: added

File: .github/workflows/extrinsic-ordering-check-from-bin.yml (hunk 4 above)
Comment (chenzongxiong) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: fixed

File: .github/workflows/extrinsic-ordering-check-from-bin.yml (hunk 5 above)
Comment (chenzongxiong) [reply .github/workflows/extrinsic-ordering-check-from-bin.yml]: removed it to simplify the script

**Comment by chenzongxiong** [comment]
> The workflow is fine, but the main problem is the comparing URL

fixed.

File: .github/workflows/create-release-draft.yml [hunk 6]
```diff
@@ -186,6 +184,73 @@ jobs:
...
+          echo "$content" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Find issues
+        uses: actions-cool/issues-helper@v3
+        id: findissueid
+        with:
+          actions: 'find-issues'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          issue-state: 'closed'
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml]: The release checklist issue will not be closed until the update is applied.

So here should be `open` or we could remove this field too.

File: .github/workflows/create-release-draft.yml [hunk 7]
```diff
@@ -186,6 +184,73 @@ jobs:
         run: |
           make clean-docker-${{ matrix.chain }}
 
+  ## check extrinsic ##
+  extrinsic-ordering-check-from-bin:
+    runs-on: self-hosted
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml]: Will it work on ubuntu-latest? If so I'd prefer that, but if it needs something from our host, we can simply leave it.

File: .github/workflows/create-release-draft.yml [hunk 8]
```diff
@@ -186,6 +184,73 @@ jobs:
...
+          RELEASE_TAG=${{ env.RELEASE_TAG }}
+          PARACHAIN_NAME=local-parachain
+          BASE_URL=ws://127.0.0.1:9944
+          chain=${{ matrix.chain }}
+          REF_URL=${{ matrix.ref_url }}
+          echo "Metadata comparison:" > output-$chain.txt
+          echo "Date: $(date)" >> output-$chain.txt
+          echo "Base: $BASE_URL" >> output-$chain.txt
+          echo "Reference: $REF_URL" >> output-$chain.txt
+          echo "Target version: $VERSION" >> output-$chain.txt
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml]: I think `VERSION` is always empty here? That's also what I saw in the example issue.

File: .github/workflows/create-release-draft.yml (hunk 7 above)
Comment (chenzongxiong) [reply .github/workflows/create-release-draft.yml]: should work as well. Let me update it.

File: .github/workflows/create-release-draft.yml (hunk 8 above)
Comment (chenzongxiong) [reply .github/workflows/create-release-draft.yml]: change to `RELEASE_TAG`

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**